core/unit_test/TestAtomicViews.hpp

```diff
@@ -112,6 +112,7 @@ struct TestViewOperator_LeftAndRight< DataType, DeviceType, 1 >
     int error_flag = 0;
 
     Kokkos::parallel_reduce( 1, driver, error_flag );
+    Kokkos::fence();
```

Comment:
Why do we need this? I guess HPX parallel_reduce is always asynchronous? The (unfortunately not fully yet written down) semantics of parallel_reduce are that it will fence if the result argument is a scalar. Only if you give it a view will it be asynchronous.